adr/ADR-42.md

```diff
@@ -0,0 +1,94 @@
+# NATS Client LifeCycle (-ERR Protocol Error Handling)
+
+| Metadata | Value          |
+| -------- | -------------- |
+| Date     | 2024-05-31     |
+| Author   | @aricart       |
+| Status   | Implemented    |
+| Tags     | client, server |
+
+## Context and Problem Statement
+
+Client lifecycle such as connect/reconnect/liveliness (ping/pong)/LDM behaviours
+are fairly complex in a NATS client. This ADR simply documents `-ERR` protocol
+messages that are sent to a client.
```

Comment:
should we expand it to other parts also? Lifecycle suggests more

Reply:
Implementing the client when you have to refer dozens of ADR's, that are related and overlapping in many places is increadibly hard.


I think that this should be part of `Connection` spec that is already in place.

Reply:
Yeah, we could rename that one to be bigger scope, agree combining is better.

Reply:
this is now integrated into the connection add